tests/js/session-tests.js

```diff
@@ -237,6 +237,114 @@ module.exports = {
             });
     },
 
+    testProgressNotificationsForRealmOpen() {
+        if (!isNodeProccess) {
+            return Promise.resolve();
+        }
+
+        const username = uuid();
+        const realmName = uuid();
+        const expectedObjectsCount = 3;
+
+        let tmpDir = tmp.dirSync();
+        let content = fs.readFileSync(__dirname + '/download-api-helper.js', 'utf8');
+        let tmpFile = tmp.fileSync({ dir: tmpDir.name });
+        fs.appendFileSync(tmpFile.fd, content, { encoding: 'utf8' });
+
+        return new Promise((resolve, reject) => {
+            //execute download-api-helper which inserts predefined number of objects into the synced realm.
+            const child = execFile('node', [tmpFile.name, username, realmName, REALM_MODULE_PATH], { cwd: tmpDir.name }, (error, stdout, stderr) => {
+                if (error) {
+                    reject(new Error('testProgressNotifications: Error executing download api helper for Realm.open ' + error));
+                }
+                resolve();
+            });
+        })
+        .then(() => {
+            return promisifiedLogin('http://localhost:9080', username, 'password').then(user => {
+                const accessTokenRefreshed = this;
+                let successCounter = 0;
+                let progressNotificationCalled = false;
+                let config = {
+                    sync: { 
+                        user, 
+                        url: `realm://localhost:9080/~/${realmName}`,
+                        _onProgress: (transferred, remaining) => { 
+                            progressNotificationCalled = true
```

Comment:
you mean to have third parameter with the expectedSize calculated to ease the user. I can add this. On our test the method is called always once with size 166. I guess it is synced at once if the realm is small